modules/glean/runtime/glean-plugin.client.ts

```diff
@@ -11,18 +13,67 @@ export default defineNuxtPlugin((nuxtApp) => {
     const devMode = useAppConfig().env === ('dev' || 'canary' || 'preview')
     const uploadEnabled = devMode // this will eventually be a setting that the user can toggle
 
-    Glean.initialize(GLEAN_APP_ID, uploadEnabled, { })
+    Glean.initialize(GLEAN_APP_ID, uploadEnabled, {})
+    userAgent.set(navigator.userAgent)
 
     // Debugging
     if (devMode) {
       Glean.setLogPings(true) // logs to the console
       Glean.setDebugViewTag('moso-elk-dev') // logs to https://debug-ping-preview.firebaseapp.com
     }
+
+    const eventListener = (event: MouseEvent) => {
+      if (event.type === 'click') {
+        handleButtonClick(event)
+        handleLinkClick(event)
+      }
+    }
+
+    function handleButtonClick(ev: MouseEvent) {
+      const eventTarget = ev?.target as Element
+      const closestButton = eventTarget.closest('button')
+
+      interface EngagementDetails {
+        [propName: string]: {
+          engagement_type: string
+          ui_additional_detail: string
+        }
+      }
+
+      const engagementDetails: EngagementDetails = {
+        'settings.interface.colorMode': {
```

Comment:
were the other 2 button events suppose to go in here too?

Reply:
I am retracting this after seeing [your new PR](https://github.com/MozillaSocial/elk/pull/60/files#diff-95b7085ba2af2d58c78f29a8b50cce435886eb1b33d9a2cc91c065dcfd0e6bbe) :D